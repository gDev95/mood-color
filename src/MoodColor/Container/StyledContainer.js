// used in order to pass prop to css 

import React from 'react'
import styled from 'styled-components'

const StyledContainer = styled.div`
display:flex;
height:400px;
justify-content: center;
align-items: center;
background: linear-gradient(to right, ${props => props.moodColorLeft}, ${props => props.moodColorRight})
`

export default StyledContainer