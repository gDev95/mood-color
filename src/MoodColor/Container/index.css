.container {
  display:flex;
  height:400px;
  justify-content: center;
  align-items: center;
  /*background-color: #1324b3; */
}