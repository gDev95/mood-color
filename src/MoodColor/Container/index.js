import React, { Component } from 'react'
import StyledContainer from './StyledContainer'
import moods from 'data.js'
import MoodSlider from '../Slider'


class Container extends Component {
	constructor(){
		super()
		this.state = {
			moodValue: 0.0,
			moodState: '',
			moodColorLeft: '',
			moodColorRight: '',
			moodFontColor: ''
		}
		this.onChange = this.onChange.bind(this)
	}
    
	onChange (value) {
		this.setState({moodValue: value})
		this.changeMood(value)
        
	}
	changeMood(moodValue) {
		moods.map((mood, index) => {		
			if (Math.round(Number(moodValue)) === index ){
				// if not mood has no meaning, do not set moodState
				if(mood.meaning)
				{
					this.setState({
						moodState: mood.meaning, 
						moodColorLeft: mood.leftColor,
						moodColorRight: mood.rightColor, 
						moodFontColor: mood.fontColor
					})
				}
				else
				{
					this.setState({
						moodColorLeft: mood.leftColor,
						moodColorRight: mood.rightColor, 
						moodFontColor: mood.fontColor
					})	
				}
				
				return index
			}
			else {
				return 0
			}
		})
	}
	render() {
		
		return (
			<StyledContainer moodColorLeft={this.state.moodColorLeft} moodColorRight={this.state.moodColorRight}>
				<MoodSlider 
					mood={this.state.moodState} 
					moodFontColor={this.state.moodFontColor} 
					moodColor={this.state.moodColor} 
					changeMood={this.onChange} />
			</StyledContainer>
		)
	}
}

export default Container