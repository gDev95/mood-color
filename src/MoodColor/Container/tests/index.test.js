import React from 'react'
import {shallow} from 'enzyme'
import toJSON from 'enzyme-to-json'
import Container from '../index'

describe('container component', () => {
	const component = shallow(
		<Container />
	)
	test('renders properly', () => {
		const tree = toJSON(component)
		expect(tree).toMatchSnapshot()
	})
	
})