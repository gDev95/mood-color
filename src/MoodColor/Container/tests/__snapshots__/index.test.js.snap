// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container component renders properly 1`] = `
<styled.div
  moodColorLeft=""
  moodColorRight=""
>
  <MoodSlider
    changeMood={[Function]}
    mood=""
    moodFontColor=""
  />
</styled.div>
`;
